vms/rpcchainvm/vm_server.go

```diff
@@ -150,40 +150,21 @@ func (vm *VMServer) Initialize(ctx context.Context, req *vmpb.InitializeRequest)
 	// Register metrics for each Go plugin processes
 	vm.processMetrics = registerer
 
-	// Dial each database in the request and construct the database manager
-	versionedDBs := make([]*manager.VersionedDatabase, len(req.DbServers))
-	for i, vDBReq := range req.DbServers {
-		version, err := version.Parse(vDBReq.Version)
-		if err != nil {
-			// Ignore closing errors to return the original error
-			_ = vm.connCloser.Close()
-			return nil, err
-		}
-
-		clientConn, err := grpcutils.Dial(
-			vDBReq.ServerAddr,
-			grpcutils.WithChainUnaryInterceptor(grpcClientMetrics.UnaryClientInterceptor()),
-			grpcutils.WithChainStreamInterceptor(grpcClientMetrics.StreamClientInterceptor()),
-		)
-		if err != nil {
-			// Ignore closing errors to return the original error
-			_ = vm.connCloser.Close()
-			return nil, err
-		}
-		vm.connCloser.Add(clientConn)
-		db := rpcdb.NewClient(rpcdbpb.NewDatabaseClient(clientConn))
-		versionedDBs[i] = &manager.VersionedDatabase{
-			Database: corruptabledb.New(db),
-			Version:  version,
-		}
-	}
-	dbManager, err := manager.NewManagerFromDBs(versionedDBs)
+	// Dial the database
+	dbclientConn, err := grpcutils.Dial(
+		req.DbServerAddr,
+		grpcutils.WithChainUnaryInterceptor(grpcClientMetrics.UnaryClientInterceptor()),
+		grpcutils.WithChainStreamInterceptor(grpcClientMetrics.StreamClientInterceptor()),
+	)
 	if err != nil {
 		// Ignore closing errors to return the original error
 		_ = vm.connCloser.Close()
```

Comment:
I think we can get rid of these lines because there's nothing to close here anymore if the db connection isn't established. Not related to this PR but I also feel like this `wrappers` utility isn't super useful now (literally just a thread-safe list of a `Closeable`) and we can consider getting rid of it since this was the only place we were using it.

Reply:
Removed `_ = vm.connCloser.Close()`